index.js

```diff
@@ -122,10 +122,12 @@ class ServerlessWebpack {
 
       'before:offline:start': () => BbPromise.bind(this)
         .then(this.validate)
+        .then(this.compile)
         .then(this.wpwatch),
       
       'before:offline:start:init': () => BbPromise.bind(this)
         .then(this.validate)
+        .then(this.compile)
         .then(this.wpwatch),
```

Comment:
* this forces a compile at start up that completes before offline exec us called
* watch remains to support manual scenarios